src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private final int startLength = 10;
+    private T[] list = (T[]) (new Object[startLength]);
+    private int arraySize = 0;
 
     @Override
     public void add(T value) {
-
+        list[arraySize] = value;
+        arraySize++;
+        sizeControl();
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index <= size() && index >= 0) {
```

Comment:
```suggestion
        if (index <= arraySize && index >= 0) {
```
в цьому випадку краще відштовхуватись одразу від змінної а не метода. Це безпечніше адже в метод може потім додаватись якась логіка

Reply:
+